include/envoy/network/io_handle.h

```diff
@@ -26,6 +26,21 @@ using RawSliceArrays = absl::FixedArray<absl::FixedArray<Buffer::RawSlice>>;
 
 namespace Network {
 
+struct Win32RedirectRecords {
+  // The size of the buffer is selected based on:
+  // https://docs.microsoft.com/en-us/windows-hardware/drivers/network/sio-query-wfp-connection-redirect-records
+  uint8_t buf_[1024];
+  unsigned long buf_size_;
+};
+
+constexpr bool win32SupportsOriginalDestination() {
+#if defined(WIN32) && defined(SIO_QUERY_WFP_CONNECTION_REDIRECT_RECORDS) && defined(SO_ORIGINAL_DST)
+  return true;
+#else
+  return false;
+#endif
+}
```

Comment:
Per my other comment, would prefer this is in a win32 dedicated file since I think it's only called from win32 guards?

Reply:
I left it as it is since we are not adding the dedicated file.

I could move it to `platform.h` if you prefer it, open to suggestions. The reason I did not do that is because changes in `platform.h` is because it destroys my inner loop. Changing `platform.h` requires to recompile almost everything.

Reply:
+1 to move this to platform. Understand on inner loop but once the code is finished seems better to move it there?

Reply:
fixed in https://github.com/envoyproxy/envoy/pull/14753/commits/414284dd3e0cbd792817258241c60e764db48bba